diff --git a/lib/query-selector/SelectorItem.js b/lib/query-selector/SelectorItem.js
index c0e807b928b0425e9052e43cd680d68ffe361cce..7f25deaa2c5d3be2e0b35e1df6140d17b17d7f8b 100644
--- a/lib/query-selector/SelectorItem.js
+++ b/lib/query-selector/SelectorItem.js
@@ -205,7 +205,8 @@ export default class SelectorItem {
                 }
                 return isFound ? { priorityWeight: 10 } : null;
             case 'checked':
-                return element[PropertySymbol.tagName] === 'INPUT' && element.checked
+                return (element[PropertySymbol.tagName] === 'INPUT' && element.checked) ||
+                    (element[PropertySymbol.tagName] === 'OPTION' && element.selected)
                     ? { priorityWeight: 10 }
                     : null;
             case 'disabled':
